express-app/controllers/answers.js

```diff
@@ -0,0 +1,139 @@
+const Respuesta = require('../models/respuesta');
+
+module.exports = {
+  /**
+  * This function changes the status of a Estudio to 'Eliminado'
+  * using the id of the estudio
+  * 
+  * 
+  * @event
+  * @param {object} request - request object 
+  * @param {object} response - response object.
+  */  
+  addAnswer: function(request, response){
+    let idPregunta = Number(request.body.id_pregunta);
+    let idSeccion = Number(request.body.id_seccion);
+    let aIndex = Number(request.body.index);
+    let idEstudio = request.body.id_estudio;
+    let valorRespuesta = request.body.answer;
+    Respuesta.find({
+      idEstudio: idEstudio,
+      idPregunta: idPregunta 
+    }, {
+      sort: 'orden'
+    }).then((respuestas) => {
+      let create = false;
+      if(respuestas.length > 0){
+        if(aIndex >= respuestas.length){
+          aIndex = respuestas.length;
+          create = true;
+        } else {
+          Respuesta.findOneAndUpdate({
+            idEstudio: idEstudio,
+            idPregunta: idPregunta,
+            orden: aIndex
+          },{
+            respuesta: valorRespuesta
+          }).then((respuesta) => {
+            return response.sendStatus(200);
+          })
+          .catch((e) => {
+            console.log(e);
+            return response.sendStatus(500);
+          });
+        }
+      }else{
+        create = true;
+      }
+      if(create){ 
+        let respuesta = Respuesta.create({
+          idEstudio: idEstudio,
+          idPregunta: idPregunta,
+          idSeccion: idSeccion,
+          orden: aIndex,
+          eleccion: null,
+          respuesta: valorRespuesta
+        });
+        respuesta.save().then((resp) => {
+          response.sendStatus(200);
+        });
+      }
+    });
+  },
+  addSelectAnswer: function(request, response){
```

Comment:
some description about what this function does, it will be easier to understand what is going on :)